adapters/huaweiads/huaweiads.go

```diff
@@ -1321,6 +1427,47 @@ func (a *adapter) extractAdmVideo(adType int32, content *content, bidType openrt
 		}
 	}
 
+	// Only for rewarded video
+	var rewardedVideoPart = ""
+	var isAddRewardedVideoPart = true
+	if adType == rewarded {
+		var staticImageUrl = ""
+		var staticImageHeight = ""
+		var staticImageWidth = ""
+		var staticImageType = "image/png"
+		if len(content.MetaData.Icon) > 0 && content.MetaData.Icon[0].Url != "" {
+			staticImageUrl = content.MetaData.Icon[0].Url
+			if content.MetaData.Icon[0].Height > 0 && content.MetaData.Icon[0].Width > 0 {
+				staticImageHeight = strconv.Itoa(int(content.MetaData.Icon[0].Height))
+				staticImageWidth = strconv.Itoa(int(content.MetaData.Icon[0].Width))
+			} else {
+				staticImageHeight = strconv.Itoa(int(adHeight))
+				staticImageWidth = strconv.Itoa(int(adWidth))
+			}
+		} else if len(content.MetaData.ImageInfo) > 0 && content.MetaData.ImageInfo[0].Url != "" {
+			staticImageUrl = content.MetaData.ImageInfo[0].Url
+			if content.MetaData.ImageInfo[0].Height > 0 && content.MetaData.ImageInfo[0].Width > 0 {
+				staticImageHeight = strconv.Itoa(int(content.MetaData.ImageInfo[0].Height))
+				staticImageWidth = strconv.Itoa(int(content.MetaData.ImageInfo[0].Width))
+			} else {
+				staticImageHeight = strconv.Itoa(int(adHeight))
+				staticImageWidth = strconv.Itoa(int(adWidth))
+			}
```

Comment:
Consider adding test coverage for this `else` block.

Reply:
done, thank you